bluemix/terminal/table.go

```diff
@@ -69,58 +75,152 @@ func (t *PrintableTable) Add(row ...string) {
 	}
 }
 
+func isWideColumn(col string) bool {
+	// list of common columns that are usually wide
+	largeColumnTypes := []string{T("ID"), T("Description")}
+
+	for _, largeColn := range largeColumnTypes {
+		if strings.Contains(largeColn, col) {
+			return true
+		}
+	}
+
+	return false
+
+}
+
+func terminalWidth() int {
+	var err error
+	terminalWidth, _, err := term.GetSize(int(os.Stdin.Fd()))
+
+	if err != nil {
+		// Assume normal 80 char width line
+		terminalWidth = 80
+	}
+	return terminalWidth
+}
+
 func (t *PrintableTable) Print() {
 	for _, row := range append(t.rows, t.headers) {
 		t.calculateMaxSize(row)
 	}
 
-	if t.headerPrinted == false {
-		t.printHeader()
-		t.headerPrinted = true
+	tbl := table.NewWriter()
+	tbl.SetOutputMirror(t.writer)
+	tbl.SuppressTrailingSpaces()
+	// remove padding from the left to keep the table aligned to the left
+	tbl.Style().Box.PaddingLeft = ""
+	tbl.Style().Box.PaddingRight = strings.Repeat(" ", minSpace)
+	// remove all border and column and row separators
+	tbl.Style().Options.DrawBorder = false
+	tbl.Style().Options.SeparateColumns = false
+	tbl.Style().Options.SeparateFooter = false
+	tbl.Style().Options.SeparateHeader = false
+	tbl.Style().Options.SeparateRows = false
+	tbl.Style().Format.Header = text.FormatDefault
+
+	headerRow, rows := t.createPrettyRowsAndHeaders()
+	columnConfig := t.createColumnConfigs()
+
+	tbl.SetColumnConfigs(columnConfig)
+	tbl.AppendHeader(headerRow)
+	tbl.AppendRows(rows)
+	tbl.Render()
+}
+
+func (t *PrintableTable) createColumnConfigs() []table.ColumnConfig {
```

Comment:
It would be good to add some more testing to explicitly try out some odd column headings to ensure there's no breaking conditions. I don't have any in mind, but would like to see if we can test some boundary conditions here.

Reply:
Added more unit tests for various scenarios.